lib/routemap.c

```diff
@@ -3256,6 +3262,8 @@ void route_map_finish(void)
 		route_map_dep_hash[i] = NULL;
 	}
 
+	hash_clean(route_map_master_hash,
```

Comment:
This memory should have been cleaned in the upper while loop at line 3254, right?

Reply:
route_map_delete function is taking care in freeing this memory.

Reply:
Are you sure? I'm blind a bit :smile:

Reply:
yeah by looking at this code, it seems to me that this code should take care of the delete via api route_map_delete:
```
/* cleanup route_map */
	while (route_map_master.head) {
		struct route_map *map = route_map_master.head;
		map->to_be_processed = false;
		route_map_delete(map);
	}
```

It is possible though that this code is not working as expected. Are you seeing a memleak via any tool?

Reply:
Here it iterates over the list, not touching the hash, right?

Reply:
route_map_free_map function does try to release hash. Are you fixing a valgrind memleak here? If we have the call stack, it will be more clear. Its okay to call hash_clean but the hash will be empty when we come at this point.
```
       hash_release(route_map_master_hash, map);
        XFREE(MTYPE_ROUTE_MAP_NAME, map->name);
        XFREE(MTYPE_ROUTE_MAP, map);
```

Reply:
Don't have a dump right now, and don't remember which test triggered this. Closing.